### PR TITLE
Gate mutation in temporal worker

### PR DESCRIPTION
**Pull request by samrabelachew**
Injects feature allocator to Github Activities so that we can support the legacy deprecation feature flag within Neptune workflows. Feature flag prevents the Dismiss and CheckRun activities from mutating PR state when legacy deprecation is not enabled.

### CODE REVIEW

File: server/lyft/feature/github_client.go [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+	appClient, err := i.ClientCreator.NewAppClient()
+	if err != nil {
+		return 0, errors.Wrap(err, "creating app client")
+	}
+
+	installation, _, err := appClient.Apps.FindOrganizationInstallation(ctx, i.Org)
+	if err != nil {
+		return 0, errors.Wrapf(err, "finding organization installation")
+	}
+	return installation.GetID(), nil
```
Comment (nishkrishnan) [inline server/lyft/feature/github_client.go:55]: why don't u reuse this code that already exists here? https://github.com/lyft/atlantis/blob/6d8d52eea9aea773737e98441e26a072b3805c7e/server/vcs/provider/github/installation.go#L18-L19

File: server/lyft/feature/github_client.go [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+	installationToken, err := c.InstallationFetcher.FetchInstallationToken(ctx)
+	if err != nil {
+		return nil, errors.Wrap(err, "fetching installation token")
+	}
+	file, err := c.FileContentsFetcher.FetchFileContents(ctx, installationToken, c.Cfg.Owner, c.Cfg.Repo, c.Cfg.Branch, c.Cfg.Path)
+	if err != nil {
+		return nil, errors.Wrap(err, "fetching file contents")
+	}
+	return file, nil
+}
```
Comment (nishkrishnan) [inline server/lyft/feature/github_client.go:105]: realistically this should be the only thing in this specific file, the rest seems generic enough to live in our github package.

File: server/neptune/workflows/activities/github.go [hunk 3]
```diff
@@ -96,6 +100,17 @@ type UpdateCheckRunResponse struct {
 }
 
 func (a *githubActivities) GithubUpdateCheckRun(ctx context.Context, request UpdateCheckRunRequest) (UpdateCheckRunResponse, error) {
+	shouldAllocate, err := a.Allocator.ShouldAllocate(feature.LegacyDeprecation, feature.FeatureContext{
+		RepoName: request.Repo.GetFullName(),
+	})
+	if err != nil {
+		return UpdateCheckRunResponse{}, errors.Wrap(err, "unable to allocate legacy deprecation feature flag")
```
Comment (nishkrishnan) [inline server/neptune/workflows/activities/github.go:107]: we shouldn't fail, just use the default value.

File: server/lyft/feature/github_client.go [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+		return nil, fmt.Errorf("got status code %d", resp.StatusCode)
+	}
+
+	decoded, err := base64.StdEncoding.DecodeString(*fileContent.Content)
+	if err != nil {
+		return nil, errors.Wrapf(err, "decoding file content")
+	}
+
+	return decoded, nil
+}
```
[the remark this answers is not in the document]
Comment (samrabelachew) [reply server/lyft/feature/github_client.go:87]: Having the feature package depend on `server/vcs/provider/github` will cause an import cycle, which is why I just duplicated and stored a small GH client here.

File: server/lyft/feature/github_client.go [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+	Repo   string
+	Branch string
+	Path   string
+}
+
+// CustomGithubRetriever uses Atlantis' internal client to retrieve the contents
+// of the feature file.  This allows us to re-use GH credentials easily as opposed
+// to the default ffclient.GithubRetriever.
+type CustomGithubRetriever struct {
+	Client githubClient
```
[the remark this answers is not in the document]
Comment (samrabelachew) [reply server/lyft/feature/github_client.go:28]: Originally, I made `CustomGithubInstallationRetriever` work for the temporalworker and gateway only, leaving legacy behind with the old system. I can migrate everything though.

File: server/lyft/feature/github_client.go (hunk 1 above)
Comment (samrabelachew) [reply server/lyft/feature/github_client.go:43]: See comment on import cycle.

File: server/lyft/feature/github_client.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (samrabelachew) [reply server/lyft/feature/github_client.go]: moved

**Review by nishkrishnan — APPROVED** [review]
Nice thanks for figuring out the import cycle.

**Pull request merged**